components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/internal/EventUtil.java

```diff
@@ -0,0 +1,220 @@
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.application.authentication.framework.exception.UserIdNotFoundException;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.common.model.ServiceProvider;
+import org.wso2.carbon.identity.event.IdentityEventConstants;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.event.Event;
+import org.wso2.carbon.identity.oauth2.model.AccessTokenDO;
+import org.wso2.carbon.identity.openidconnect.internal.OpenIDConnectServiceComponentHolder;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Utility class for publishing OAuth related events.
+ * This class provides methods to publish token revoke events with various parameters.
+ * It uses the Identity Event Service to handle the event publication.
+ */
+public class EventUtil {
+
+    public static final Log LOG = LogFactory.getLog(EventUtil.class);
+
+    private static final String CONSUMER_KEYS = "CONSUMER_KEYS";
+    private static final String EVENT_CAUSE = "EVENT_CAUSE";
+
+    /**
+     * Publishes a token revoke event with the specified consumer keys and user details.
+     *
+     * @param consumerKeys Set of consumer keys associated with the revoked tokens.
+     * @param user         Authenticated user whose tokens are revoked.
+     * @param username     Username of the authenticated user.
+     */
+    public static void publishTokenRevokeEvent(Set<String> consumerKeys, AuthenticatedUser user, String username) {
+
+        Map<String, Object> properties = baseProperties(user);
+        properties.put(CONSUMER_KEYS, consumerKeys);
+        publish(properties);
+    }
+
+    /**
+     * Publishes a token revoke event with the specified access token details.
+     *
+     * @param token AccessTokenDO object containing details of the revoked token.
+     */
+    public static void publishTokenRevokeEvent(AccessTokenDO token) {
+
+        if (token == null) {
+            return;
+        }
+
+        Map<String, Object> properties = baseProperties(token.getAuthzUser());
+        properties.put(CONSUMER_KEYS, Collections.singletonList(token.getConsumerKey()));
+        publish(properties);
+    }
+
+    /**
+     * Publishes a token revoke event with the specified application ID,
+     * consumer key, removed scopes, and tenant domain.
+     *
+     * @param applicationResourceId Application ID associated with the revoked tokens.
+     * @param consumerKey           Consumer key of the application.
+     * @param removedScopes         List of scopes that were removed.
+     * @param tenantDomain          Tenant domain of the application.
+     */
+    public static void publishTokenRevokeEvent(String applicationResourceId, String consumerKey,
```

Comment:
This method have 4 params, what happen if we need to include one more param in the event. Then we need to introduce another method and deprecate this.

Rather can we have a generic method with EventSubType, Properties (Map) as param